[PRD-5563] - Bold font in windows menu

RPAraujo opened the pull request:
Cleaned up SystemInformation.java
Fixed bug and added unit tests

wingman-pentaho commented:
### Build Failed

:x: Something went wrong while validating this pull request.

#### Build Commands

```sh
ant -Dtestreports.dir=bin/reports/unit-test -f libraries/libbase/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f libraries/libbase/build.xml jacoco-integration checkstyle publish-local
```

##### Stdout log
###### (last 100 lines)

```console

compile.res_copy:

compile.lic_copy:

copy-license-lgpl:

copy-license-gpl:

compile.post:

compile:

clean-tests:
   [delete] Deleting directory /home/buildguy/project/libraries/libbase/bin/test

junit.exists-check:
     [echo] junit.available=true

init-tests:
    [mkdir] Created dir: /home/buildguy/project/libraries/libbase/bin/test/classes

compile-tests:
    [javac] /home/buildguy/project/libraries/libbase/build-res/subfloor.xml:1773: warning: 'includeantruntime' was not set, defaulting to build.sysclasspath=last; set to false for repeatable builds
    [javac] Compiling 15 source files to /home/buildguy/project/libraries/libbase/bin/test/classes
     [copy] Copying 1 file to /home/buildguy/project/libraries/libbase/bin/test/classes

init-test-reports:
    [mkdir] Created dir: /home/buildguy/project/libraries/libbase/bin/reports/integration-test/xml
    [mkdir] Created dir: /home/buildguy/project/libraries/libbase/bin/reports/integration-test/html

jacoco-integration:
    [mkdir] Created dir: /home/buildguy/project/libraries/libbase/bin/reports/jacoco-integration
[jacoco:coverage] Enhancing junit with coverage
[junitreport] Processing /home/buildguy/project/libraries/libbase/bin/reports/integration-test/html/TESTS-TestSuites.xml to /tmp/null1050433024
[junitreport] Loading stylesheet jar:file:/home/buildguy/apache-ant-1.9.7/lib/ant-junit.jar!/org/apache/tools/ant/taskdefs/optional/junit/xsl/junit-frames.xsl
[junitreport] Transform time: 672ms
[junitreport] Deleting: /tmp/null1050433024

antcontrib.download-check:

antcontrib.download:

install-antcontrib:

subfloor.install-ivy:
     [echo] Skipping IVY install.  IVY has already been configured by the build

install-ivy:

install-checkstyle:
[ivy:resolve] :: Apache Ivy 2.4.0 - 20141213170938 :: http://ant.apache.org/ivy/ ::
[ivy:resolve] :: loading settings :: url = file:/home/buildguy/project/libraries/libbase/ivysettings.xml
[ivy:resolve] :: resolving dependencies :: #;working@01fbc2631421
[ivy:resolve] 	confs: [default]
[ivy:resolve] 	found com.puppycrawl.tools#checkstyle;6.12 in pentaho-mvn
[ivy:resolve] 	found org.apache.commons#commons-lang3;3.4 in pentaho-mvn
[ivy:resolve] 	found antlr#antlr;2.7.7 in pentaho-mvn
[ivy:resolve] 	found org.antlr#antlr4-runtime;4.5.1-1 in pentaho-mvn
[ivy:resolve] 	found commons-beanutils#commons-beanutils;1.9.2 in pentaho-mvn
[ivy:resolve] 	found commons-logging#commons-logging;1.1.1 in pentaho-mvn
[ivy:resolve] 	found commons-collections#commons-collections;3.2.1 in pentaho-mvn
[ivy:resolve] 	found commons-cli#commons-cli;1.3.1 in pentaho-mvn
[ivy:resolve] 	found com.google.guava#guava;18.0 in pentaho-mvn
[ivy:resolve] 	found org.apache.ant#ant;1.9.6 in pentaho-mvn
[ivy:resolve] 	found org.apache.ant#ant-launcher;1.9.6 in pentaho-mvn
[ivy:resolve] :: resolution report :: resolve 771ms :: artifacts dl 24ms
	---------------------------------------------------------------------
	|                  |            modules            ||   artifacts   |
	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
	---------------------------------------------------------------------
	|      default     |   11  |   0   |   0   |   0   ||   11  |   0   |
	---------------------------------------------------------------------
[ivy:retrieve] :: retrieving :: #
[ivy:retrieve] 	confs: [default]
[ivy:retrieve] 	11 artifacts copied, 0 already retrieved (7336kB/536ms)

checkstyle.xml-report:
      [get] Getting: http://nexus.pentaho.org/content/groups/omni/com/pentaho/pentaho-coding-standards/pentaho-checkstyle/1.0.5/pentaho-checkstyle-1.0.5.xml
      [get] To: /home/buildguy/project/libraries/libbase/checkstyle/pentaho-checkstyle-1.0.5.xml
      [get] Getting: http://nexus.pentaho.org/content/groups/omni/com/pentaho/pentaho-coding-standards/pentaho-checkstyle-suppressions/1.0.5/pentaho-checkstyle-suppressions-1.0.5.xml
      [get] To: /home/buildguy/project/libraries/libbase/checkstyle/suppressions.xml
    [mkdir] Created dir: /home/buildguy/project/libraries/libbase/checkstyle/reports/xml
[checkstyle] Running Checkstyle 6.12 on 101 files

checkstyle.html-report:
    [mkdir] Created dir: /home/buildguy/project/libraries/libbase/checkstyle/reports/html
      [get] Getting: http://nexus.pentaho.org/content/groups/omni/omni/com/pentaho/pentaho-coding-standards/pentaho-checkstyle-stylesheet/1.0.5/pentaho-checkstyle-stylesheet-1.0.5.xsl
      [get] To: /home/buildguy/project/libraries/libbase/checkstyle/pentaho-checkstyle-stylesheet-1.0.5.xsl
      [get] Error opening connection java.io.FileNotFoundException: http://nexus.pentaho.org/content/groups/omni/omni/com/pentaho/pentaho-coding-standards/pentaho-checkstyle-stylesheet/1.0.5/pentaho-checkstyle-stylesheet-1.0.5.xsl
      [get] Error opening connection java.io.FileNotFoundException: http://nexus.pentaho.org/content/groups/omni/omni/com/pentaho/pentaho-coding-standards/pentaho-checkstyle-stylesheet/1.0.5/pentaho-checkstyle-stylesheet-1.0.5.xsl
      [get] Error opening connection java.io.FileNotFoundException: http://nexus.pentaho.org/content/groups/omni/omni/com/pentaho/pentaho-coding-standards/pentaho-checkstyle-stylesheet/1.0.5/pentaho-checkstyle-stylesheet-1.0.5.xsl
      [get] Can't get http://nexus.pentaho.org/content/groups/omni/omni/com/pentaho/pentaho-coding-standards/pentaho-checkstyle-stylesheet/1.0.5/pentaho-checkstyle-stylesheet-1.0.5.xsl to /home/buildguy/project/libraries/libbase/checkstyle/pentaho-checkstyle-stylesheet-1.0.5.xsl

BUILD FAILED
/home/buildguy/project/libraries/libbase/build-res/subfloor.xml:2796: Can't get http://nexus.pentaho.org/content/groups/omni/omni/com/pentaho/pentaho-coding-standards/pentaho-checkstyle-stylesheet/1.0.5/pentaho-checkstyle-stylesheet-1.0.5.xsl to /home/buildguy/project/libraries/libbase/checkstyle/pentaho-checkstyle-stylesheet-1.0.5.xsl

Total time: 10 seconds
rm: cannot remove ‘/home/buildguy/.ivy2/local’: No such file or directory
no publish local to remove
```

#### Stderr log
###### (last 100 lines)

```console
java.lang.Exception: org.pentaho.build.buddy.bundles.orchestrator.OrchestratorImpl$BuildException: java.io.IOException: Head build returned with code 1
	at org.pentaho.build.buddy.bundles.orchestrator.OrchestratorImpl.orchestrate(OrchestratorImpl.java:363)
	at Proxyff41e6fb_6e3d_40fa_bffd_63e02e15b250.orchestrate(Unknown Source)
	at org.pentaho.build.buddy.bundles.rest.OrchestratorRestService$1.write(OrchestratorRestService.java:62)
	at org.apache.cxf.jaxrs.provider.BinaryDataProvider.writeTo(BinaryDataProvider.java:172)
	at org.apache.cxf.jaxrs.utils.JAXRSUtils.writeMessageBody(JAXRSUtils.java:1381)
	at org.apache.cxf.jaxrs.interceptor.JAXRSOutInterceptor.serializeMessage(JAXRSOutInterceptor.java:244)
	at org.apache.cxf.jaxrs.interceptor.JAXRSOutInterceptor.processResponse(JAXRSOutInterceptor.java:120)
	at org.apache.cxf.jaxrs.interceptor.JAXRSOutInterceptor.handleMessage(JAXRSOutInterceptor.java:83)
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308)
	at org.apache.cxf.interceptor.OutgoingChainInterceptor.handleMessage(OutgoingChainInterceptor.java:83)
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308)
	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121)
	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:253)
	at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234)
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208)
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160)
	at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:180)
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:298)
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doPost(AbstractHTTPServlet.java:217)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:273)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:812)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:587)
	at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:70)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:271)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:80)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:499)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:311)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:544)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.pentaho.build.buddy.bundles.orchestrator.OrchestratorImpl$BuildException: java.io.IOException: Head build returned with code 1
	at org.pentaho.build.buddy.bundles.orchestrator.OrchestratorImpl.doOrchestrate(OrchestratorImpl.java:578)
	at org.pentaho.build.buddy.bundles.orchestrator.OrchestratorImpl.orchestrate(OrchestratorImpl.java:345)
	... 41 more
Caused by: java.io.IOException: Head build returned with code 1
	... 43 more

```

RPAraujo commented:
@pamval, @pedrofvteixeira Plase review
I opted for keeping two commits, instead of squashing them, for better readability of the changes made

pedrofvteixeira commented:
One commit for checkstyle and license header cleanup, and another with the fix alone, correct ?
I'm good with these changes.  

@pamval, your call.

pamval commented on libraries/libbase/source/org/pentaho/reporting/libraries/base/util/SystemInformation.java:
```diff
@@ -32,13 +32,30 @@
-    if ( logger.isInfoEnabled() ) {
-      logger.info( getSystemInformation() );
+    if ( getLogger().isInfoEnabled() ) {
+      getLogger().info( getOtherProperties() );
```
Shouldn't this call getSystemInformation as before?

RPAraujo replied:
Yes it should! Sorry for the slip-up. I must have changed it inadvertently during the refactoring. I'll fix it.

pamval commented on libraries/libbase/source/org/pentaho/reporting/libraries/base/util/SystemInformation.java:
```diff
@@ -89,9 +106,12 @@ private String formatLongPathList( String value ) {
     return value;
   }
 
+  public Map<String, String> getOtherPropertiesInst() {
```
Is this method used anywhere?

RPAraujo replied:
No! Sorry, its leftover trash from when we where developing unit tests. I'll remove it.

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
ant -Dtestreports.dir=bin/reports/unit-test -f libraries/libbase/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f libraries/libbase/build.xml jacoco-integration checkstyle publish-local
```

#### Cleanup Commands
```sh
rm -r ~/.ivy2/local || echo no publish local to remove
```

#### Changed files

```console
  libraries/libbase/source/org/pentaho/reporting/libraries/base/util/SystemInformation.java
  libraries/libbase/test-src/org/pentaho/reporting/libraries/base/util/SystemInformationTest.java
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.reporting.libraries.base.util.SystemInformation
 * Branch Change:  + 58.33%
 * Complexity Change:  + 50.00%
 * Instruction Change:  + 36.53%
 * Line Change:  + 28.46%
 * Method Change:  + 50.00%

#### Checkstyle violations

The code submitted is not complying with one or more of our guidelines. To perform these checks locally, [please use this tool](https://github.com/pentaho/pentaho-coding-standards).

###### /home/buildguy/project/libraries/libbase/test-src/org/pentaho/reporting/libraries/base/util/SystemInformationTest.java

> :heavy_exclamation_mark: Unused import - org.junit.Assert.assertNotNull. at Line: 26 Column: 15

> :heavy_exclamation_mark: Unused import - org.junit.Assert.assertTrue. at Line: 27 Column: 15

> :heavy_exclamation_mark: ')' is not preceded with whitespace. at Line: 58 Column: 47

pamval commented:
@RPAraujo check style violations. PLease review

pamval commented:
And also please squash the two commits with the fix into one.

RPAraujo commented:
@pamval, @pedrofvteixeira please review
Squashed the commits and added Powermockito to unit test the static method, this allowed removing the global static variable.

wingman-pentaho commented:
### Build Completed

:white_check_mark: This pull request has passed all validations.

#### Build Commands

```sh
ant -Dtestreports.dir=bin/reports/unit-test -f libraries/libbase/build.xml clean-all resolve jacoco && ant -Dtestreports.dir=bin/reports/integration-test -f libraries/libbase/build.xml jacoco-integration checkstyle publish-local
```

#### Cleanup Commands
```sh
rm -r ~/.ivy2/local || echo no publish local to remove
```

#### Changed files

```console
  libraries/libbase/ivy.xml
  libraries/libbase/source/org/pentaho/reporting/libraries/base/util/SystemInformation.java
  libraries/libbase/test-src/org/pentaho/reporting/libraries/base/util/SystemInformationTest.java
```

tmorgner commented:
@pamval  @RPAraujo 

Sorry, dont agree with that fix. You are hijacking a logging method to do initialization. That is wrong. 

If you want to run code for report-designer when it boots up, then run it in ReportDesignerBoot or in the report-designer Main class. Apparently something in UIManager.getLookAndFeel is triggering the changed behaviour - so please find out which code exactly it is and then move it to the "ReportDesigner" class.

There is a "SetLookAndFeelTask" that is a much better place to hold this L&F initialisation.

pamval commented:
@RPAraujo can you please look at this again, following Thomas comments?
